### PR TITLE
Fix various warnings

### PR DESCRIPTION
**Pull request by jaredsburrows**
task -> task.register
all -> configureEach

### CODE REVIEW

File: gradle-versions-plugin/build.gradle [hunk 1]
```diff
@@ -41,14 +50,16 @@ dependencies {
...
+  dependsOn dokkaHtml
   group = 'Publications'
   description = 'Create jar of documentation.'
   archiveClassifier = 'javadoc'
   from dokkaJavadoc.outputDirectory
 }
 
-task sourcesJar(type: Jar, dependsOn: classes) {
+tasks.register('sourcesJar', Jar) {
+  dependsOn classes
```
Comment (ben-manes) [inline gradle-versions-plugin/build.gradle:62]: I think nowadays one can use `java.withSourcesJar()`

File: gradle-versions-plugin/build.gradle [hunk 2]
```diff
@@ -41,14 +50,16 @@ dependencies {
   testImplementation 'org.spockframework:spock-junit4:2.1-M2-groovy-3.0', { exclude module: 'groovy-all' } // Use localGroovy()
 }
 
-task docsJar(type: Jar, dependsOn: dokkaHtml) {
+tasks.register('docsJar', Jar) {
+  dependsOn dokkaHtml
```
Comment (ben-manes) [inline gradle-versions-plugin/build.gradle:54]: and here I think can be done via `java.withJavadocJar()`

Comment (jaredsburrows) [reply gradle-versions-plugin/build.gradle:54]: I think we have this custom task for dokka.

File: gradle-versions-plugin/build.gradle (hunk 1 above)
Comment (jaredsburrows) [reply gradle-versions-plugin/build.gradle:62]: I can move us to use https://github.com/vanniktech/gradle-maven-publish-plugin to help remove these custom tasks in the future.

**Review by ben-manes — APPROVED** [review]

File: gradle-versions-plugin/build.gradle (hunk 1 above)
Comment (jaredsburrows) [reply gradle-versions-plugin/build.gradle:53]: @ben-manes Actually, we are only using Gradle's publish plugin - https://docs.gradle.org/current/userguide/publishing_gradle_plugins.html we are not publishing the docs or sources. Looks like we have not published to maven in a long time? https://mvnrepository.com/artifact/com.github.ben-manes/gradle-versions-plugin

Comment (ben-manes) [reply gradle-versions-plugin/build.gradle:53]: Oh right, it predated the plugin portal by a few years.

**Pull request merged**